code/game/machinery/medical_kiosk.dm

```diff
@@ -246,9 +246,8 @@
 				var/bit_vol = bit.volume - belly.food_reagents[bit.type]
 				if(bit_vol > 0)
 					chemical_list += list(list("name" = bit.name, "volume" = round(bit_vol, 0.01)))
-	if(altPatient.reagents.addiction_list.len)
-		for(var/datum/reagent/R in altPatient.reagents.addiction_list)
-			addict_list += list(list("name" = R.name))
+	for(var/datum/reagent/R in altPatient.reagents?.addiction_list)
```

Comment:
Is this how ?. works? from minor testing and looking at the doc, ?. doesn't care if the var accessed is null, it only cares about the object we're accessing from. See http://www.byond.com/docs/ref/#/operator/?.

Reply:
Ack, this one is me being dumb and experimenting with stuff with deleting the initial if and forgetting to rework after it still worked. The ? should be before the .len as below.

Reply:
what is the ?. supposed to do here

also this should be typeless, the only items that should be in this list are reagents

Reply:
See <https://hackmd.io/@tgstation/ry6DxGvDL#Performance-of-untyped-vs-typed-loops> for more details, the ? should be unneeded, as reagents should always exist, unless we're really fucked

Reply:
The ?. was based on the assumption that since reagents is initially initialised as null, it would _maybe_ make sense to check? If it's unneeded I'll remove them.